CBinding/CMake/CMakeProject.cs

```diff
@@ -247,32 +142,26 @@ void AddFile (FilePath fileName, string targetName)
 			}
 		}
 
-		protected override Task<BuildResult> OnBuild (ProgressMonitor monitor, ConfigurationSelector configuration,
+		protected async override Task<BuildResult> OnBuild (ProgressMonitor monitor, ConfigurationSelector configuration,
 													  OperationContext operationContext)
 		{
-			return Task.Factory.StartNew (() => {
-				BuildResult results;
+			BuildResult results;
 
-				if (!CheckCMake ()) {
-					results = new BuildResult ();
-					results.AddError ("CMake cannot be found.");
-					return results;
-				}
+			if (!CheckCMake ()) {
+				results = new BuildResult ();
+				results.AddError ("CMake cannot be found.");
+				return results;
+			}
 
-				FileService.CreateDirectory (file.ParentDirectory.Combine (outputDirectory));
+			FileService.CreateDirectory (file.ParentDirectory.Combine (outputDirectory));
 
-				monitor.BeginStep ("Generating build files.");
-				Stream generationResult = ExecuteCommand ("cmake", "../", outputDirectory, monitor);
-				results = ParseGenerationResult (generationResult, monitor);
-				monitor.EndStep ();
+			DefaultToolchain = PropertyService.Get<string> ("CBinding.DefaultToolchain", new MinGW32Toolchain ().Name);
```

Comment:
I would suggest moving the "property & default toolchain logic to `CMakeToolchain.GetToolchain()`. currently you have it replicated in two places.

The default should also be platform dependent.